ray-operator/apis/ray/v1/rayservice_types.go

```diff
@@ -57,6 +66,9 @@ type RayServiceSpec struct {
 	DeploymentUnhealthySecondThreshold *int32 `json:"deploymentUnhealthySecondThreshold,omitempty"`
 	// ServeService is the Kubernetes service for head node and worker nodes who have healthy http proxy to serve traffics.
 	ServeService *corev1.Service `json:"serveService,omitempty"`
+	// UpgradeStrategy represents the strategy used when upgrading the RayService. Currently supports `NewCluster` and `None`
+	// +kubebuilder:default:=NewCluster
+	UpgradeStrategy RayServiceUpgradeStrategy `json:"upgradeStrategy,omitempty"`
```

Comment:
what will happen if specify a value for `UpgradeStrategy` other than `NewCluster` or `None`?

Reply:
Added validation to validateRayServiceSpec